src/urllib3/util/request.py

```diff
@@ -103,9 +114,7 @@ def make_headers(
     return headers
 
 
-def set_file_position(
-    body: Any, pos: Optional[Union[int, object]]
-) -> Optional[Union[int, object]]:
+def set_file_position(body: Any, pos: _TYPE_BODY_POSITION) -> _TYPE_BODY_POSITION:
```

Comment:
This would now be `Optional[_TYPE_BODY_POSITION]`. It returns a non-Optional `_TYPE_BODY_POSITION`.

Reply:
This won't work with following error, as we are trying to assign our sentinel `_FAILEDTELL` to pos.

```
error: Incompatible types in assignment (expression has type "_TYPE_FAILEDTELL", variable has type "Optional[int]")  [assignment]
```

Reply:
I think we need to keep `Optional[_TYPE_BODY_POSITION]` only instead of `Optional[int]` to keep it uniform.

Yes, `urlopen` should not take `_FAILEDTELL`, but it is taking the `body_pos` and typing wise above solution is not bad.